go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -880,33 +916,46 @@ func commonFlows(ofPortPhys, bridgeMacAddress, ofPortPatch, ofPortHost string) [
 			defaultOpenFlowCookie, ofPortPhys, bridgeMacAddress, ofPortPatch, ofPortHost))
 
 	if config.IPv4Mode {
-		// table 0, packets coming from pods headed externally. Commit connections
+		// table 0, packets coming from pods headed externally. Commit connections with ct_mark ctMarkOVN
 		// so that reverse direction goes back to the pods.
 		dftFlows = append(dftFlows,
 			fmt.Sprintf("cookie=%s, priority=100, in_port=%s, ip, "+
-				"actions=ct(commit, zone=%d), output:%s",
-				defaultOpenFlowCookie, ofPortPatch, config.Default.ConntrackZone, ofPortPhys))
+				"actions=ct(commit, zone=%d, exec(set_field:%s->ct_mark)), output:%s",
+				defaultOpenFlowCookie, ofPortPatch, config.Default.ConntrackZone, ctMarkOVN, ofPortPhys))
+
+		// table 0, packets coming from host Commit connections with ct_mark ctMarkHost
+		// so that reverse direction goes back to the host.
+		dftFlows = append(dftFlows,
+			fmt.Sprintf("cookie=%s, priority=100, in_port=%s, ip, "+
```

Comment:
seems like this should be OK, because the external traffic policy uses priority 110 and filters on specific tp src/dst:
https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/node/gateway_shared_intf.go#L130

those should take higher precedence than these flows, and the shared gw masquerade flows are a 500, also taking precedence over these